modules/auxiliary/scanner/nntp/nntp_login.rb

```diff
@@ -0,0 +1,189 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'NNTP Login Utility',
+      'Description' => %q{
+        This module attempts to authenticate to NNTP services
+        which support the AUTHINFO authentication extension.
+
+        This module supports AUTHINFO USER/PASS authentication,
+        but does not support AUTHINFO GENERIC or AUTHINFO SASL
+        authentication methods.
+      },
+      'Author'      => 'Brendan Coles <bcoles[at]gmail.com>',
+      'License'     => MSF_LICENSE,
+      'References'  => [ [ 'CVE', '1999-0502' ], # Weak password
+                         [ 'URL', 'https://tools.ietf.org/html/rfc3977' ],
+                         [ 'URL', 'https://tools.ietf.org/html/rfc4642' ],
+                         [ 'URL', 'https://tools.ietf.org/html/rfc4643' ] ]))
+    register_options(
+      [
+        Opt::RPORT(119),
+        OptPath.new('USER_FILE', [ false, 'The file that contains a list of probable usernames.',
+          File.join(Msf::Config.install_root, 'data', 'wordlists', 'unix_users.txt') ]),
+        OptPath.new('PASS_FILE', [ false, 'The file that contains a list of probable passwords.',
+          File.join(Msf::Config.install_root, 'data', 'wordlists', 'unix_passwords.txt') ])
+      ])
+    deregister_options 'RHOST'
+  end
+
+  def run_host(ip)
+    begin
+      connect
+      return :abort unless nntp?
+      return :abort unless supports_authinfo?
+
+      report_service :host  => rhost,
+                     :port  => rport,
+                     :proto => 'tcp',
+                     :name  => 'nntp'
+      disconnect
+
+      each_user_pass { |user, pass| do_login user, pass }
+    rescue ::Interrupt
+      raise $ERROR_INFO
+    rescue EOFError, ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout
+      print_error "#{peer} Connection failed"
+      return
+    rescue OpenSSL::SSL::SSLError => e
+      print_error "SSL negotiation failed: #{e}"
+    rescue => e
+      print_error "#{peer} Error: #{e.class} #{e} #{e.backtrace}"
+      return
+    ensure
+      disconnect
+    end
+  end
+
+  def nntp?
+    banner = sock.get_once
+
+    if !banner
+      vprint_error "#{peer} No response"
+      return false
+    end
+
+    if banner !~ /^200/
+      print_error 'Unexpected reply'
```

Comment:
IMO, here and anywhere you are printing some sort of status output, include `peer`.  Otherwise the error messages are lacking context, right?